### ISSUE TITLE
Django CMS appears to be incompatible with custom AdminSite classes

### ISSUE THREAD

**Issue by mikemcgowan** [divio/django-cms#4103]
No Django CMS admin URLs can be reversed when a [custom AdminSite class](
https://docs.djangoproject.com/en/1.7/ref/contrib/admin/#customizing-the-adminsite-class) is present.

The issue is described in full [here](https://github.com/mikemcgowan/djangocms-custom-adminsite).

**Comment by yakky** [comment]
There is an undocumented configuration variable to set the custom admin namespace: ``CMS_ADMIN_NAMESPACE``
What if you set that?

**Comment by mikemcgowan** [comment]
Thanks for the reply, I tried setting `CMS_ADMIN_NAMESPACE = 'fooadmin'` but it wasn't clear to me what else (if anything) I should have been setting. It didn't work.

I searched the CMS source in my virtualenv and found that that setting is only used in the function `admin_reverse` and it's unclear to me under what circumstances that function is called.

What I *did* manage to find was that I could use my custom admin as a *second* custom `AdminSite` (as described [here](https://docs.djangoproject.com/en/1.7/ref/contrib/admin/#multiple-admin-sites-in-the-same-urlconf)) without anything obviously breaking.
